pkgs/top-level/all-packages.nix

```diff
@@ -27163,9 +27163,7 @@ with pkgs;
 
   bonzomatic = callPackage ../applications/editors/bonzomatic { };
 
-  bottles = callPackage ../applications/misc/bottles {
-    wine = wineWowPackages.minimal;
```

Comment:
Can this be kept, but null by default (or something similar), so that one can easily override bottles if wine is required?

Reply:
I don't really get the point of being able to do that, because the point of Bottles is to use the Wine versions downloaded from the app, not the system one.
If someone wanted to use system Wine, it works with a Wine installed via `environment.systemPackages`.
And if someone _really_ needs to use a specific version of system Wine just for Bottles (which, I repeat, is probably downloadable from Bottles itself), they could still do `(bottles.overrideAttrs (oldAttrs: rec {propagatedBuildInputs = oldAttrs.propagatedBuildInputs ++ [ myWine ];}))`.

Reply:
> I don't really get the point of being able to do that, because the point of Bottles is to use the Wine versions downloaded from the app, not the system one. If someone wanted to use system Wine, it works with a Wine installed via `environment.systemPackages`. And if someone _really_ needs to use a specific version of system Wine just for Bottles (which, I repeat, is probably downloadable from Bottles itself), they could still do `(bottles.overrideAttrs (oldAttrs: rec {propagatedBuildInputs = oldAttrs.propagatedBuildInputs ++ [ myWine ];}))`.

I don't really like to have wine as a system package in my case and anyway the nix provided wine can still be useful to someone, for example the bottles vaniglia(stock wine) runner is not currently updated to wine 7.19, and they don't update them very frequently, also the nix provided wine is different from the bottles provided one because it includes many [dependencies](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/applications/emulators/wine/base.nix) that can be useful to run different software. In general you can have more compatibility just by enabling one package.

Also, the official flatpak does include wine by default and the aur package makes it optional.
Considering that it can be useful, and it is listed as an optional dependency by bottles, it makes sense to at least have it as an optional option that doesn't require too much thinkering for the user to be enabled.

Reply:
Would it be better to also change its name (for example, from `wine` to something like `winePackage`)?
So that there is no "name conflict", and instead of needing to override it from `all-packages.nix` we can simply have `winePackage ? null`.

Reply:
> Would it be better to also change its name (for example, from `wine` to something like `winePackage`)? So that there is no "name conflict", and instead of needing to override it from `all-packages.nix` we can simply have `winePackage ? null`.

I am not sure, but lutris is overwritten like this in `all-packages.nix`: 
```
  lutris-unwrapped = python3.pkgs.callPackage ../applications/misc/lutris {
    wine = wineWowPackages.staging;
  };
  lutris = callPackage ../applications/misc/lutris/fhsenv.nix {
    buildFHSUserEnv = buildFHSUserEnvBubblewrap;
  };
  lutris-free = lutris.override {
    steamSupport = false;
  };
```